src/tests/architect_tests/test_planner.py

```diff
@@ -7,106 +7,129 @@
+        features_by_table={"features2": ["f3", "f4"], "features3": ["f5", "f6"]},
     )
     feature_dicts = [feature_dict_one, feature_dict_two]
     planner = Planner(
         feature_start_time=datetime.datetime(2010, 1, 1, 0, 0),
-        label_names=['booking'],
-        label_types=['binary'],
-        cohort_name='prior_bookings',
-        states=['state_one AND state_two'],
+        label_names=["booking"],
+        label_types=["binary"],
+        cohort_name="prior_bookings",
+        states=["state_one AND state_two"],
         user_metadata={},
     )
 
-    updated_matrix_definitions, build_tasks = \
-        planner.generate_plans(matrix_set_definitions, feature_dicts)
+    updated_matrix_definitions, build_tasks = planner.generate_plans(
+        matrix_set_definitions, feature_dicts
+    )
     # test that it added uuids: we don't much care what they are
     matrix_uuids = []
     for matrix_def in updated_matrix_definitions:
-        assert isinstance(matrix_def['train_uuid'], str)
-        matrix_uuids.append(matrix_def['train_uuid'])
-        for test_uuid in matrix_def['test_uuids']:
+        assert isinstance(matrix_def["train_uuid"], str)
+        matrix_uuids.append(matrix_def["train_uuid"])
+        for test_uuid in matrix_def["test_uuids"]:
             assert isinstance(test_uuid, str)
     assert len(set(matrix_uuids)) == 4
 
     # not going to assert anything on the keys (uuids), just get out the values
     build_tasks = build_tasks.values()
     assert len(build_tasks) == 8  # 2 splits * 2 matrices per split * 2 feature dicts
 
-    assert sum(1 for task in build_tasks if task['matrix_type'] == 'train') == 4
-    assert sum(1 for task in build_tasks if task['matrix_type'] == 'test') == 4
-    assert sum(1 for task in build_tasks if task['feature_dictionary'] == feature_dict_one) == 4
-    assert sum(1 for task in build_tasks if task['feature_dictionary'] == feature_dict_two) == 4
-    assert sum(
-        1 for task in build_tasks
-        if task['matrix_metadata']['feature_groups'] == ['first_features']
-    ) == 4
-    assert sum(
-        1 for task in build_tasks
-        if task['matrix_metadata']['feature_groups'] == ['second_features']
-    ) == 4
-    assert sum(
-        1 for task in build_tasks
-        if task['matrix_metadata']['cohort_name'] == 'prior_bookings'
-    ) == 8
+    assert sum(1 for task in build_tasks if task["matrix_type"] == "train") == 4
+    assert sum(1 for task in build_tasks if task["matrix_type"] == "test") == 4
+    assert (
+        sum(1 for task in build_tasks if task["feature_dictionary"] == feature_dict_one)
+        == 4
+    )
+    assert (
+        sum(1 for task in build_tasks if task["feature_dictionary"] == feature_dict_two)
+        == 4
+    )
+    assert (
+        sum(
+            1
+            for task in build_tasks
+            if task["matrix_metadata"]["feature_groups"] == ["first_features"]
+        )
+        == 4
+    )
```

Comment:
I don't imagine we can ask more of an automated tool, here, but yikes.

Reply:
Yeah. Looks kind of weird, but I never liked the way I formatted these lines before either (put the == 4 next to the prior end-paren) so :man_shrugging: